datafusion/physical-expr/src/math_expressions.rs

```diff
@@ -226,21 +227,23 @@ pub fn round(args: &[ArrayRef]) -> Result<ArrayRef> {
                     }
                 )) as ArrayRef)
             }
-            ColumnarValue::Array(decimal_places) => Ok(Arc::new(make_function_inputs2!(
```

Comment:
I basically give up using the existing macro (`make_function_inputs2`), not sure is it worth changing it to be possibly return `Result` within the `$FUNC`.

Reply:
Nice work! This overflow issue can be solved during binding in the future(hopefully...), I think it's not necessary to change the existing macro 🤔